### PR TITLE
Remove Tasks wrapped in a Task

### PR DESCRIPTION
**Pull request by Bond-009**
https://github.com/jellyfin/jellyfin/compare/master...Bond-009:tasks?expand=1#diff-4ee36df3f7cae4f4687c920c303b76e7R55
Example where returning the task is better than awaiting it.
Also cleaned up some warnings and useless methods.

### CODE REVIEW

File: Emby.Server.Implementations/ScheduledTasks/ScheduledTaskWorker.cs [hunk 1]
```diff
@@ -360,7 +360,7 @@ async void trigger_Triggered(object sender, EventArgs e)
         /// <exception cref="InvalidOperationException">Cannot execute a Task that is already running</exception>
         public async Task Execute(TaskOptions options)
         {
-            var task = Task.Run(async () => await ExecuteInternal(options).ConfigureAwait(false));
+            var task = ExecuteInternal(options);
```
Comment (cvium) [inline Emby.Server.Implementations/ScheduledTasks/ScheduledTaskWorker.cs]: Didn't the old line actually start up a new thread?

File: MediaBrowser.Controller/Net/BasePeriodicWebSocketListener.cs [hunk 2]
```diff
@@ -98,11 +93,6 @@ private void Start(WebSocketMessageInfo message)
             var dueTimeMs = long.Parse(vals[0], UsCulture);
             var periodMs = long.Parse(vals[1], UsCulture);
 
-            if (vals.Length > 2)
-            {
-                ParseMessageParams(vals.Skip(2).ToArray());
```
Comment (JustAMan) [inline MediaBrowser.Controller/Net/BasePeriodicWebSocketListener.cs]: are you sure no subclasses override this function?

Comment (Bond-009) [reply MediaBrowser.Controller/Net/BasePeriodicWebSocketListener.cs]: Yep, just double checked.

**Comment by cvium** [comment]
Large parts of this PR conflicts with #1010.

**Comment by joshuaboniface** [comment]
@Bond-009 Still several conflicts, can you resolve or is this one unnecessary now?

File: MediaBrowser.Api/Library/LibraryService.cs [hunk 3]
```diff
@@ -988,19 +988,16 @@ private int GetCount(Type type, User user, GetItemCounts request)
         /// Posts the specified request.
         /// </summary>
         /// <param name="request">The request.</param>
-        public void Post(RefreshLibrary request)
+        public async Task Post(RefreshLibrary request)
         {
-            Task.Run(() =>
```
Comment (cvium) [inline MediaBrowser.Api/Library/LibraryService.cs]: I'm still not convinced this won't be slower

Comment (Bond-009) [reply MediaBrowser.Api/Library/LibraryService.cs]: Only 1 way to know for sure ;)

**Pull request merged**